tests/bdd/environment.py

```diff
@@ -1,15 +1,22 @@
 """Entry point and hooks for behave."""
 
+from pkg_resources import resource_filename
+
 from lms.app import create_app
+from tests.bdd.higher_order_gherkin import Injector
 from tests.bdd.step_context import StepContextManager
 from tests.conftest import TEST_SETTINGS
 
 TEST_SETTINGS["session_cookie_secret"] = "notasecret"
 
+# Create the compiled step file before steps are read
+Injector.create_step_file(
+    source_dir=resource_filename("tests", "bdd/steps/feature_steps/"),
+    target_file=resource_filename("tests", "bdd/steps/_compiled_feature_steps.py"),
```

Comment:
Is there any concern with this output file getting particularly long, or with diffs across commits being a little cumbersome?

Reply:
It might get long, but I'd recommend we view it as compiled code and not review it (like a pyc file). If the system works, then we should trust it.

One change I thought would be nice is to layout the files in the same structure they were found with. This would make it easier to mentally map from one to the other, but this might be a future improvement.